portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +32,50 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+    
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> comments = new ArrayList<>();
+
+    String commentLimitString = request.getParameter("limit");
+    int commentLimit = Integer.parseInt(commentLimitString);
```

Comment:
You might consider having a default value to fall back on in case the request doesn't include one.

Reply:
Now I see that you have the minimum value of 5 below, but still what would happen if the request didn't include the `limit` parameter?

Reply:
Done.